rofl-client/rs/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+        })
+        .await
+        .map_err(|e| -> Box<dyn std::error::Error> { Box::new(e) })?
+        .map_err(|e| -> Box<dyn std::error::Error> { Box::new(e) })?;
+        Ok(res)
+    }
+
+    // POST /rofl/v1/tx/sign-submit
+    pub async fn sign_submit(
+        &self,
+        tx: Tx,
+        encrypt: Option<bool>,
+    ) -> Result<String, Box<dyn std::error::Error>> {
+        let sock = self.socket_path.clone();
+        let req = serde_json::to_vec(&SignSubmitRequest { tx, encrypt })?;
+        let res = tokio::task::spawn_blocking(move || -> std::io::Result<String> {
+            let body = http_unix_request(
+                &sock,
+                "POST",
+                "/rofl/v1/tx/sign-submit",
+                Some(&req),
+                Some("application/json"),
+            )?;
+            let resp: SignSubmitResponse = serde_json::from_slice(&body)
+                .map_err(|e| std::io::Error::new(std::io::ErrorKind::InvalidData, e))?;
+            Ok(resp.data)
+        })
+        .await
+        .map_err(|e| -> Box<dyn std::error::Error> { Box::new(e) })?
+        .map_err(|e| -> Box<dyn std::error::Error> { Box::new(e) })?;
+        Ok(res)
+    }
+
+    /// Convenience helper for ETH-style call
+    pub async fn sign_submit_eth(
+        &self,
+        gas_limit: u64,
+        to: &str,
+        value: u64,
+        data_hex: &str,
+        encrypt: Option<bool>,
+    ) -> Result<String, Box<dyn std::error::Error>> {
+        let eth = EthCall {
+            gas_limit,
+            to: to.to_string(),
+            value,
+            data: data_hex.to_string(),
+        };
+        self.sign_submit(Tx::Eth(eth), encrypt).await
+    }
+}
+
+// Blocking HTTP-over-UDS request using std::os::unix::net::UnixStream
+fn http_unix_request(
+    socket_path: &str,
+    method: &str,
+    path: &str,
+    body: Option<&[u8]>,
+    content_type: Option<&str>,
+) -> std::io::Result<Vec<u8>> {
+    use std::{
+        io::{Error, ErrorKind, Read, Write},
+        os::unix::net::UnixStream,
+    };
+
+    let mut stream = UnixStream::connect(socket_path)?;
+
+    let mut req = Vec::new();
+    req.extend_from_slice(format!("{method} {path} HTTP/1.1\r\n").as_bytes());
+    req.extend_from_slice(b"Host: localhost\r\n");
+    req.extend_from_slice(b"Connection: close\r\n");
+    if let Some(ct) = content_type {
+        req.extend_from_slice(format!("Content-Type: {ct}\r\n").as_bytes());
+    }
+    if let Some(b) = body {
+        req.extend_from_slice(format!("Content-Length: {}\r\n", b.len()).as_bytes());
+    }
+    req.extend_from_slice(b"\r\n");
+    if let Some(b) = body {
+        req.extend_from_slice(b);
+    }
+
+    stream.write_all(&req)?;
+    stream.flush()?;
+
+    let mut resp = Vec::new();
+    let mut buf = [0u8; 8192];
+    loop {
+        let n = stream.read(&mut buf)?;
+        if n == 0 {
+            break;
+        }
+        resp.extend_from_slice(&buf[..n]);
+    }
+
+    let header_end = resp
```

Comment:
Would it be a bit easier to maintain if we used something like https://crates.io/crates/httparse?